samples/cpp/visual_language_chat/visual_language_chat.cpp

```diff
@@ -13,7 +16,40 @@ int main(int argc, char* argv[]) try {
     if (3 != argc) {
         throw std::runtime_error(std::string{"Usage "} + argv[0] + " <MODEL_DIR> <IMAGE_FILE>");
     }
-    ov::Tensor image = utils::load_image(argv[2]);
+
+    // multinomial or beam_search can be used as well
+    ov::genai::GenerationConfig generation_config = ov::genai::greedy();
+    // ov::genai::GenerationConfig generation_config = ov::genai::multinomial();
+    // ov::genai::GenerationConfig generation_config = ov::genai::beam_search();
+
+    ov::AnyMap properies;
+    properies.insert(ov::genai::generation_config(generation_config));
+
+    // streamer could be used with greedy and multinomial
+     // if num_return_sequences > 1 in case of multinomial, the streamer will use the output from the first sequence
+    if (generation_config.is_greedy_decoding() or generation_config.is_multinomial()) {
+        properies.insert(ov::genai::streamer(print_subword));
+    }
+
+    std::vector<ov::Tensor> images;
+    std::string input_path = argv[2];
+    if (!input_path.empty() && fs::exists(input_path)) {
+        if (fs::is_directory(input_path)) {
+            for (const auto& dir_entry : fs::directory_iterator(input_path)) {
+                ov::Tensor image = utils::load_image(dir_entry.path());
+                images.push_back(std::move(image));
+            }
+        } else if (fs::is_regular_file(input_path)) {
+            ov::Tensor image = utils::load_image(input_path);
+            images.push_back(std::move(image));
+        }
+    }
+
+    if (images.empty())
+        throw std::runtime_error("No one image found by path " + input_path);
+    else
+        properies.insert(images.size() == 1 ? ov::genai::image(images.at(0)) : ov::genai::images(images));
```

Comment:
Nice idea processing the folder. Please, extend the test to run the sample with a folder as well.
Factor folder processing out to a function similar to `utils::load_image()` so we don't overload the sample file.

Reply:
maybe we should always use `ov::genai::images(images)` as it covers single image case?

Reply:
Having `image` isn't a big burden. But in case of this sample, having `images` only would simplify the implementation, which is a good thing.